framework/platform/bpl/uci/db/bpl_db_uci.h

```diff
@@ -0,0 +1,134 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * SPDX-FileCopyrightText: 2020 the prplMesh contributors (see AUTHORS.md)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#ifndef _BPL_DB_UCI_H_
+#define _BPL_DB_UCI_H_
+
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace beerocks {
+namespace bpl {
+namespace db {
+
+/****************************************************************************/
+/******************************* Definitions ********************************/
+/****************************************************************************/
+
+/****************************************************************************/
+/******************************* Structures *********************************/
+/****************************************************************************/
+
+/****************************************************************************/
+/******************************** Functions *********************************/
+/****************************************************************************/
+
+/**
+ * @brief Find if entry exists.
+ * 
+ * @param[in] config_file name of the requested configuration.
+ * @param[in] entry_type type of the requested entry.
+ * @param[in] entry_name name of the requested entry.
+ * @return true if entry exists in db, false otherwise
+ */
+bool uci_entry_exists(const std::string &config_file, const std::string &entry_type,
+                      const std::string &entry_name);
+
+/**
+ * @brief Add new named entry.
+ * 
+ * @param[in] config_file name of the requested configuration.
+ * @param[in] entry_type type of the requested entry.
+ * @param[in] entry_name name of the requested entry.
+ * @return true if entry exists in db, false otherwise
+ */
+bool uci_add_entry(const std::string &config_file, const std::string &entry_type,
+                   const std::string &entry_name);
+
+/**
+ * @brief Set values in entry, updating as needed.
+ * A parameter set to an empty string will remove it from the entry.
+ * A parameter that does not exist will be added to the entry.
+ * A parameter that already exists will be overridden with the new value
```

Comment:
```suggestion
 * @brief Set values in entry, updating as needed.
 *
 * An option set to an empty string will remove it from the section.
 * An option that does not exist will be added to the section.
 * An option that already exists will be overridden with the new value
 * Options that already exist and are not mentioned in @a params will remain unchanged.
```

I.e. param -> option and entry -> section. Same in other places.

Reply:
Renamed to match UCI names